modules/markup/markdown/toc.go

```diff
@@ -0,0 +1,49 @@
+// Copyright 2020 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package markdown
+
+import (
+	"fmt"
+	"net/url"
+
+	"github.com/unknwon/i18n"
+	"github.com/yuin/goldmark/ast"
+)
+
+func createTOCNode(toc []Header, lang string) ast.Node {
+	details := NewDetails()
+	summary := NewSummary()
+
+	summary.AppendChild(summary, ast.NewString([]byte(i18n.Tr(lang, "toc"))))
+	details.AppendChild(details, summary)
+	ul := ast.NewList('-')
+	details.AppendChild(details, ul)
+	currentLevel := 6
+	for _, header := range toc {
+		if header.Level < currentLevel {
+			currentLevel = header.Level
+		}
+	}
+	for _, header := range toc {
+		if header.Level < currentLevel {
+			ul = ul.Parent().(*ast.List)
```

Comment:
Shouldn't it be more like?:
```suggestion
			for header.Level < currentLevel {
				ul = ul.Parent().(*ast.List)
				currentLevel--
			}
```

Reply:
yup. Dunno what is up with me...